### DISCUSSION TITLE
Achieving Data Locality - Running steps on same node

### DISCUSSION THREAD

**Discussion by itajaja** [argoproj/argo-workflows#5236] [Q&A] [answered]
Is there a way in Argo to specify that steps in the workflow need to run on the same node, so that they can share a hostpath volume? I am not sure how I could use affinity to achieve that with just kubernetes out of the box.

**Comment by terrytangyuan** [comment, marked as answer]
Yes, you can specify `nodeSelector` and `affinity` at either workflow or template level. Here's an example of using node selector in a step: https://github.com/argoproj/argo-workflows/blob/master/examples/node-selector.yaml

**Reply by itajaja** [reply]
thanks. but I am not sure you can use nodeSelector to point specifically to a node vs to a collection of nodes with same label

**Reply by alexec** [reply]
I think you can use pod-affinity to do this.

**Reply by itajaja** [reply]
Yes, you guys are totally right. I think colocation is somewhat orthogonal to Argo. thanks!
